cmd/juju/application/deployer/bundlehandler.go

```diff
@@ -396,23 +395,16 @@ func (h *bundleHandler) resolveCharmChannelAndRevision(charmURL string, charmBas
 	if h.isLocalCharm(charmURL) {
 		return charmChannel, -1, nil
 	}
-	// If the charm URL already contains a revision, return that before
-	// attempting to resolve a revision from any charm repository. We can ignore the
-	// error here, as we want to just parse out the charm URL.
-	// Resolution and validation of the charm URL happens further down.
-	if curl, err := charm.ParseURL(charmURL); err == nil {
-		if charm.Local.Matches(curl.Schema) {
-			return charmChannel, -1, nil
-		} else if curl.Revision >= 0 && charmChannel != "" {
-			return charmChannel, curl.Revision, nil
-		}
-	}
```

Comment:
No point ignoring the error parsing here, since the next step was to parse again. (This is why we don't ignore the error anymore)

More broadly, there is no point parsing the url twice